playwright/stream.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import io
+from typing import Dict
+
+from playwright.connection import ChannelOwner
+
+
+class Stream(ChannelOwner):
+    def __init__(
+        self, parent: ChannelOwner, type: str, guid: str, initializer: Dict
+    ) -> None:
+        super().__init__(parent, type, guid, initializer)
+
+    async def stream(self, fp: io.BytesIO) -> io.BytesIO:
+        if not fp:
+            fp = io.BytesIO()
+
+        while True:
+            result = await self._channel.send("read", dict(size=1))
```

Comment:
Are you reading single byte at a time? Probably not too efficient!

Reply:
Oh ye good catch you won! Currently its not possible to leave out the size, thats why I set it to Node.js default value.

Probably we have to mark it in the protocol as optional so Node.js alone handles the default value?